core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -65,13 +88,55 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
       partitions = infos.map(_.toTopicPartition)
       _ <- consumer.assign(partitions)
       _ <- partitions.traverse_(
-        tp => offsets.get(tp).map(o => consumer.seek(tp, o)).getOrElse(seekTo(consumer, List(tp)))
+        tp =>
+          offsets
+            .get(tp)
+            .map(o => consumer.seek(tp, o))
+            .getOrElse(SeekTo.seek(consumer, List(tp), seekTo))
       )
     } yield ()
 
   def assign(topic: String, offsets: Map[TopicPartition, Long])(implicit F: Monad[F]): F[Unit] =
     assign(List(topic), offsets)
 
+  def seekToCommittedOr(
+      partitions: Set[TopicPartition],
+      seekTo: SeekTo,
+  )(implicit F: Monad[F]): F[Unit] =
+    for {
+      committed <- consumer.committed(partitions)
+      _ <- partitions.toList.traverse_(
+        tp =>
+          committed
+            .get(tp)
+            //p could be mapped to an explicit null value
+            .flatMap(Option(_))
+            .fold(SeekTo.seek(consumer, List(tp), seekTo))(o => consumer.seek(tp, o.offset))
+      )
+    } yield ()
+
+  def assignAndSeekToCommittedOr(
+      topics: List[String],
+      seekTo: SeekTo,
+  )(implicit F: Monad[F]): F[Unit] =
+    for {
+      infos <- consumer.partitionsFor(topics)
+      partitions = infos.map(_.toTopicPartition)
+      () <- consumer.assign(partitions)
+      () <- seekToCommittedOr(partitions.toSet, seekTo)
+    } yield ()
+
+  def subscribeAndSeekToCommittedOr(
```

Comment:
Method names this complex are always a smell to me, but I also know you're head is much more into this than mine. I wonder if we should do a huddle to walk through this PR.